src/Tools/dotnet-monitor/DiagnosticsMonitorCommandHandler.cs

```diff
@@ -20,60 +21,70 @@ namespace Microsoft.Diagnostics.Tools.Monitor
 {
     internal sealed class DiagnosticsMonitorCommandHandler
     {
-        public async Task<int> Start(CancellationToken token, IConsole console, string[] urls)
+        public async Task<int> Start(CancellationToken token, IConsole console, string[] urls, bool metrics)
         {
             //CONSIDER The console logger uses the standard AddConsole, and therefore disregards IConsole.
-            using IWebHost host = CreateWebHostBuilder(console, urls).Build();
+            using IWebHost host = CreateWebHostBuilder(console, urls, metrics).Build();
             await host.RunAsync(token);
             return 0;
         }
 
-        public IWebHostBuilder CreateWebHostBuilder(IConsole console, string[] urls)
+        public IWebHostBuilder CreateWebHostBuilder(IConsole console, string[] urls, bool metrics)
         {
+            string metricsEndpoint = null;
+            if (metrics)
+            {
+                metricsEndpoint = GetMetricsEndpoint();
+                urls = new List<string>(urls) { metricsEndpoint }.ToArray();
+            }
+
             IWebHostBuilder builder = WebHost.CreateDefaultBuilder()
                 .ConfigureAppConfiguration((IConfigurationBuilder builder) =>
                 {
-                    ConfigureNames(builder);
+                    if (metrics)
+                    {
+                        ConfigureMetricsEndpoint(builder, metricsEndpoint);
+                    }
                 })
                 .ConfigureServices((WebHostBuilderContext context, IServiceCollection services) =>
                 {
                     //TODO Many of these service additions should be done through extension methods
                     services.AddSingleton<IDiagnosticServices, DiagnosticServices>();
-                    services.Configure<ContextConfiguration>(context.Configuration);
+                    if (metrics)
+                    {
+                        services.Configure<PrometheusConfiguration>(context.Configuration.GetSection(nameof(PrometheusConfiguration)));
+                    }
                 })
                 .UseUrls(urls)
                 .UseStartup<Startup>();
 
             return builder;
         }
 
-        private static void ConfigureNames(IConfigurationBuilder builder)
+        private string GetMetricsEndpoint()
         {
-            string hostName = Environment.GetEnvironmentVariable("HOSTNAME");
-            if (string.IsNullOrEmpty(hostName))
-            {
-                hostName = Environment.MachineName;
-            }
-            string namespaceName = null;
-            try
-            {
-                string nsFile = @"/var/run/secrets/kubernetes.io/serviceaccount/namespace";
-                if (File.Exists(nsFile))
-                {
-                    namespaceName = File.ReadAllText(nsFile);
-                }
-            }
-            catch
+            string endpoint = "http://localhost:52325";
```

Comment:
These values can the be the default, but shouldn't they also be configurable at the command line? Just like the -u/--urls option?

Reply:
Yep, I have a TODO to expand on the metrics configuration, including things like https endpoints (since prometheus does support them).